packages/components/src/plugins/index.js

```diff
@@ -52,36 +53,36 @@ export class Plugins extends Component {
 			return false;
 		}
 
-		const installs = await Promise.all(
-			pluginSlugs.map( async ( slug ) => {
-				return await installPlugin( slug );
-			} )
-		);
-
-		const installErrors = installs.filter(
-			( install ) => install.status !== 'success'
-		);
+		const installs = await installPlugins( pluginSlugs );
 
-		if ( installErrors.length ) {
-			this.handleErrors( installErrors );
+		if ( installs.errors && Object.keys( installs.errors.errors ).length ) {
+			this.handleErrors( installs.errors.errors );
 			return;
 		}
 
 		const activations = await activatePlugins( pluginSlugs );
 
-		if ( activations.status === 'success' ) {
-			this.handleSuccess( activations.activePlugins );
+		if ( activations.success && activations.data.activated ) {
+			this.handleSuccess( activations.data.activated );
 			return;
 		}
 
-		this.handleErrors( activations );
+		this.handleErrors( activations.errors.errors );
 	}
 
 	handleErrors( errors ) {
 		const { onError, createNotice } = this.props;
 
-		errors.forEach( ( error ) => {
-			createNotice( 'error', error );
+		Object.keys( errors ).forEach( ( plugin ) => {
+			createNotice(
+				'error',
+				pluginNames[ plugin ]
+					? errors[ plugin ][ 0 ].replace(
+							`\`${ plugin }\``,
+							pluginNames[ plugin ]
+					  )
+					: errors[ plugin ][ 0 ]
+			);
 		} );
 
 		this.setState( { hasErrors: true } );
```

Comment:
I noticed a bug here, unrelated to these changes: In the render function, `hasErrors` is gathered from props, not state, so the "Retry" or "Continue without installing" options never get displayed when errors occur.

Reply:
Great catch.   Went ahead and updated in https://github.com/woocommerce/woocommerce-admin/pull/4411/commits/9d24050f07eee59955444fd654c82811b21af6a6.